tasks/dataset-split/main.py

```diff
@@ -0,0 +1,129 @@
+            splitDataset.add(sample.zipPath)
+
+        splitDatasetsList.append(splitDataset)
+
+    return splitDatasetsList
+
+
+def sequenceDatasetSplit(splitSamples: list[list[SequenceSample]], metadata: Any, projectID: int) -> list[CustomDataset]:
+    metadataAddress = list(metadata.load().folderContent)[0] #address where the file metadata is located in the form of a string
+    
+    with open(metadataAddress, mode="r", newline="") as file:
+        reader = csv.DictReader(file)
+        originalMetadata: list[dict] = []
+        
+        for row in reader:
+            originalMetadata.append(dict(row))
+
+    splitDatasetsList: list[CustomDataset] = []
+    
+    for index, sampleChunk in enumerate(splitSamples):
+        splitDataset = CustomDataset.createDataset(f"{currentTaskRun().id}-newDataset-{index}", projectID)
+        splitMetadataList: list[dict] = []
+        
+        for sample in sampleChunk:
+            sample.unzip()
+            splitDataset.add(sample.zipPath)
+            
+            fieldNames = list(originalMetadata[0].keys())
+            
+            for oneMetadata in originalMetadata:
+                if sample.name.startswith(oneMetadata[fieldNames[0]].split("_")[0]):
+                    splitMetadataList.append(oneMetadata)
+
+        metadataCSV = f"_metadata_{index}.csv"
+        with open(metadataCSV, "w", newline="") as file:
+            writer = csv.DictWriter(file, fieldnames=fieldNames)
+            writer.writeheader()
+            writer.writerows(splitMetadataList)
+
+        metadataZIP = f"_metadata_{index}.zip"
+        with zipfile.ZipFile(metadataZIP, "w") as zipFile:
+            zipFile.write(metadataCSV)
+
+        splitDataset.add(metadataZIP)
+
+    splitDatasetsList.append(splitDataset)
+          
+    return splitDatasetsList
+
+
+def main() -> None:
+    taskRun = currentTaskRun()
+    originalDataset = taskRun.dataset
+    
+    newDatasetCount = taskRun.parameters["numberOfNewDatasets"]
+    projectID = taskRun.projectId
+
+    n = originalDataset.count
+    if n <= newDatasetCount or newDatasetCount < 2:
+        logging.error("Number of samples is smaller than the number you want to divide the database")
+        sys.exit("The End")
+
+    splitSamples: list[list[SampleType]] = samplesSplit(originalDataset, newDatasetCount)
+
+    splitDatasetsList: list[DatasetType]
+
+    if isinstance(originalDataset, ImageDataset):
+        splitDatasetsList = imageDatasetSplit(splitSamples, originalDataset.classes, projectID)
+
+    if isinstance(originalDataset, CustomDataset):
+        try:
+            taskRun.setDatasetType(SequenceDataset)
+            originalDataset = taskRun.dataset
+            splitSamples = samplesSplit(originalDataset, newDatasetCount)
+            listOfNewDatasets = sequenceDatasetSplit(splitSamples, originalDataset.metadata, projectID)
+        except:
+            splitDatasetsList = customDatasetSplit(splitSamples, projectID) 
```

Comment:
Generated datasets need to be submitted as output at the end of the execution if everything was succesful.